### PR TITLE
fix: adjusting container instrumentation infrastructure monitoring to highlight assignment of container logs

### PR DESCRIPTION
**Pull request by harrykimpel**
Please follow [conventional commit standards](https://www.conventionalcommits.org/en/v1.0.0/)
in your commit messages and pull request title.

## Give us some context

* What problems does this PR solve?
This PR will highlight some required changes in order to assign container logs to our container entities rather than the host entity

* Add any context that will help us review your changes such as testing notes,
  links to related docs, screenshots, etc.
there are some threads in our forum (see [here](https://forum.newrelic.com/s/hubtopic/aAX8W0000008cwxWAA/no-logs-on-a-container-page) and [here](https://forum.newrelic.com/s/hubtopic/aAX8W0000008dEZWAY/new-relic-one-container-data-logs-not-appearing-in-the-explorer-containers))

* If your issue relates to an existing GitHub issue, please link to it.
n/a

### CODE REVIEW

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/container-instrumentation-infrastructure-monitoring-logs.mdx
Comment (akristen) [inline src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/container-instrumentation-infrastructure-monitoring-logs.mdx:19]: Do customers need to have instrumented their containers? Should we instead link them to the other containers dock as a prerequisite? I imagine they can't collect logs about containers w/o having instrumentation first. 

@harrykimpel

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/container-instrumentation-infrastructure-monitoring.mdx
Comment (akristen) [inline src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/container-instrumentation-infrastructure-monitoring.mdx:2]: @harrykimpel How do feel about making this title clearer by renaming it to "Monitor your containers with the infrastructure agent"?

File: src/nav/infrastructure.yml
Comment (akristen) [inline src/nav/infrastructure.yml:55]: Let's delete this title once we confirm how we want to frame it. I think "Monitor containers" is slightly more accurate than "instrument your host containers" (since the doc does say we don't need to do anything extra

**Comment by akristen** [comment]
leaving a note for myself to update the URL (or revert that change) once i get SME approval from Core team

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/container-instrumentation-infrastructure-monitoring-logs.mdx
Comment (harrykimpel) [reply src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/container-instrumentation-infrastructure-monitoring-logs.mdx]: @akristen yes, that is correct. I guess it makes sense tzo link them to the other containers doc.

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/container-instrumentation-infrastructure-monitoring.mdx
Comment (harrykimpel) [reply src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/container-instrumentation-infrastructure-monitoring.mdx]: @akristen agreed, sounds great!

**Review by harrykimpel — COMMENTED** [review]
LGTM!

File: src/nav/infrastructure.yml
Comment (harrykimpel) [reply src/nav/infrastructure.yml:55]: agree

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx
Comment (DavSanchez) [inline src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx:17]: The Docker Compose mention is not 100% accurate. The containers do not need to be running as part of a Docker Compose setup, they could have been ran  'manually' and the infrastructure-agent would be able to pick the logs up all the same.

Comment (harrykimpel) [reply src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx:17]: I just tested with a plain Docker container running manually using 'docker run ...' and the same applies. I need to run `sudo docker run --log-opt tag="{{.Name}}" -d log-generator` in order to get that tag added as attribute.

I will update the docs page accordingly.

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx
Comment (jsubirat) [inline src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx:34]:
```suggestion
1. Create a file under `/etc/newrelic-infra/logging.d/docker-containers.yml` with the following contents:

```yml
logs:
    - name: docker_containers
      file: /var/lib/docker/containers/*/*.log
```
```

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx
Comment (jsubirat) [inline src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx:43]: We probably don't need to add this. Please note that the Log Forwarder integration picks any YAML file under `/etc/newrelic-infra/logging.d/*.yml` and translates them to Fluent Bit configuration. Contrarily to the public belief, they don't need to be placed under a single `/etc/newrelic-infra/logging.d/logging.yml` file. I believe this missconception was caused by Virtuoso, because that is the file it is using.

In this case, I think it makes sense to just use a separate file to include the Docker container logs config. It results in simpler installation instructions as well, IMHO, as we don't need to tell the user "edit the file in this particular way" (error prone, specially when editing YAML files), but just "create a file with these exact contents".

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx
Comment (jsubirat) [inline src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx:71]: Is this `attrs.tag` added in the client side (that is, before sending the logs to New Relic)? Or is this metadata information sent by the Infra-Agent and later on used by the Logs Pipeline to enrich the logs?

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx
Comment (jsubirat) [inline src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx:83]: What is `**HERE**`? Did you enable some way to somehow correlate the logs to a container instead of to a host?

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx
Comment (jsubirat) [inline src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx:103]: I'm confused here. The Infra-Agent log forwarder normally gets the `entity.guid` belonging to the _host_ running these containers, so the logs get associated to the host. Are you doing something differently in this case, when running Docker containers? My understanding is that by having `containerName` in the logs, you are somehow leveraging some Entity Synthesis rule to associate the log to a particular container. And I assume this is happening because the Infra-Agent actually sends `entity.guid.INFRA`, and not `entity.guid`. So by the time entity synthesis runs, `entity.guid` is still not populated.

I'm afraid that this solution is making some assumptions in the order in which the backend will perform the actions (that is, first it will execute parsing, then entity synthesis, and later on entity.guid consolidation), so I'd probably advise against this. I'd not like to have this coupling in the pipeline, as it may block us from being able to evolve it and modify the step order if we ever need to.

PS: have you verified that if logs get associated to the container they don't get associated to the host _as well_?

**Review by jsubirat — CHANGES_REQUESTED** [review]
Hey @harrykimpel, I think there could be a safer (and easier) alternative here, to ensure that logs are cleanly associated with a container and they do _not_ get associated with a host. We could modify the Fluent Bit configuration generated by the Infra-Agent so that it removes the `entity.guid.INFRA` of the host if it detects that it comes from a Docker container, and then just renames `attrs.tag` into `containerName`. I think we could (and should) implement this as code, and do it out-of-the-box in the Infra-Agent. Please let me discuss this internally and will get back to you. For the time being, I'd like to ask this to wait a little bit before getting merged.

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx
Comment (DavSanchez) [reply src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx]: Hi @harrykimpel , sorry I didn't notice this was specific to Docker Compose. Thanks for clarifying!

Comment (harrykimpel) [reply src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx]: > Hi @harrykimpel , sorry I didn't notice this was specific to Docker Compose. Thanks for clarifying!

@DavSanchez it was actually a great comment since I was not aware that the same concept also works for manually running Docker containers.

**Comment by akristen** [comment]
I'm adding this doc to sprint starting 29 APR

File: src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx
Comment (akristen) [reply src/content/docs/infrastructure/install-infrastructure-agent/linux-installation/enable-logs-monitored-containers.mdx]: Added to the PR!

**Comment by akristen** [comment]
Closing this PR until @jsubirat 's team notifies us with a better solution to complete this action.

**Pull request closed without merging**
